python_modules/libraries/dagster-aws/dagster_aws/s3/io_manager.py

```diff
@@ -100,11 +104,75 @@ def handle_output(self, context, obj):
         context.add_output_metadata({"uri": MetadataValue.path(path)})
 
 
+class PickledObjectS3IOManagerResource(ConfigurableIOManager):
+    """Persistent IO manager using S3 for storage.
+
+    Serializes objects via pickling. Suitable for objects storage for distributed executors, so long
+    as each execution node has network connectivity and credentials for S3 and the backing bucket.
+
+    Assigns each op output to a unique filepath containing run ID, step key, and output name.
+    Assigns each asset to a single filesystem path, at "<base_dir>/<asset_key>". If the asset key
+    has multiple components, the final component is used as the name of the file, and the preceding
+    components as parent directories under the base_dir.
+
+    Subsequent materializations of an asset will overwrite previous materializations of that asset.
+    With a base directory of "/my/base/path", an asset with key
+    `AssetKey(["one", "two", "three"])` would be stored in a file called "three" in a directory
+    with path "/my/base/path/one/two/".
+
+    Example usage:
+
+    .. code-block:: python
+
+        from dagster import asset, Definitions
+        from dagster_aws.s3 import PickledObjectS3IOManagerResource, S3Resource
+
+
+        @asset
+        def asset1():
+            # create df ...
+            return df
+
+        @asset
+        def asset2(asset1):
+            return asset1[:5]
+
+        defs = Definitions(
+            assets=[asset1, asset2],
+            resources={
+                "io_manager": PickledObjectS3IOManagerResource(
+                    s3_resource=S3Resource(),
+                    s3_bucket="my-cool-bucket",
+                    s3_prefix="my-cool-prefix",
+                )
+            }
+        )
+
+    """
+
+    s3_resource: ResourceDependency[S3Resource]
+    s3_bucket: str = Field(description="S3 bucket to use for the file manager.")
+    s3_prefix: str = Field(
+        default="dagster", description="Prefix to use for the S3 bucket for this file manager."
+    )
+
+    @cached_method
+    def inner_db_io_manager(self) -> PickledObjectS3IOManager:
+        return PickledObjectS3IOManager(
+            s3_bucket=self.s3_bucket,
+            s3_session=self.s3_resource,
+            s3_prefix=self.s3_prefix,
+        )
+
+    def load_input(self, context: InputContext) -> Any:
+        return self.inner_db_io_manager().load_input(context)
+
+    def handle_output(self, context: OutputContext, obj: Any) -> None:
+        return self.inner_db_io_manager().handle_output(context, obj)
+
+
```

Comment:
@sryza this is still definitely making me wince. I think this will be copied and pasted throughout the land and it's quite a bit of boilerplate, in the end it's extremely similar behavior.

Definitely still think we should support people making IOManagers directly configurable but forcing us to do these delegation shenanigans rather than just support factory directly seems like a lot.

Reply:
Regardless of the wisdom of invoking list_objects as a health check, its there and removing it is a large behavior change that will potentially break users if we remove it.